src/components/ComputeModal.js

```diff
@@ -899,8 +914,13 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
                   id,
                   isSearchable: false,
                   value: currentNumCpus,
-                  onChange: ({ value }) => updateComputeConfig('masterMachineType',
-                    _.find({ cpu: value }, validMachineTypes)?.name || mainMachineType),
+                  onChange: ({ value }) => {
+                    const mainMachineType = getMainMachineTypeByNumCpus(value)
+                    const { validGpuType: newGpuType, validNumGpus: newNumGpus } = getValidCpuGpuConfig(mainMachineType)
+                    updateComputeConfig('masterMachineType', mainMachineType)
+                    updateComputeConfig('gpuType', newGpuType)
+                    updateComputeConfig('numGpus', newNumGpus)
```

Comment:
These two lines are what fix the bug where we weren't updating the GPU config accordingly to the newly selected machine type. Similarly for the other `Select` below.

Reply:
Considered creating a composite out of `gpuType` and `numGpus` but there are two places where they don't get updated at the same time so decided to keep the current way.